### PR TITLE
improve risk acceptance UI (accept aditional findings)

### PR DESCRIPTION
**Pull request by FelixHernandez**
## Improve risk acceptance UI (accept aditional findings)
[sc-3336]

**Description**

this feature imporves the UI in risk_acceptance view in the part where the user accepts aditional findings. Now a table is shown when the user can select multiple findings and search or filter by multiple fields.

**Test results**
image

### CODE REVIEW

**Review by mtesauro — APPROVED** [review]
Approved

File: dojo/engagement/views.py [hunk 1]
```diff
@@ -1144,11 +1142,16 @@ def view_edit_risk_acceptance(request, eid, raid, edit_mode=False):
 
     unaccepted_findings = Finding.objects.filter(test__in=eng.test_set.all(), risk_accepted=False) \
         .exclude(id__in=accepted_findings).order_by("title")
-    add_fpage = get_page_items(request, unaccepted_findings, 10, 'apage')
+    add_fpage = get_page_items(request, unaccepted_findings, 500, 'apage')
```
Comment (Maffooch) [inline dojo/engagement/views.py:1145]:
```suggestion
    add_fpage = get_page_items(request, unaccepted_findings, 25, 'apage')
```
500 seems excessive. Most other pages show 25 findings

File: dojo/forms.py [hunk 2]
```diff
@@ -774,15 +775,17 @@ class Meta:
...
 class AddFindingsRiskAcceptanceForm(forms.ModelForm):
+
     accepted_findings = forms.ModelMultipleChoiceField(
-        queryset=Finding.objects.none(), required=True,
-        widget=forms.widgets.SelectMultiple(attrs={'size': 10}),
-        help_text=('Select to add findings.'), label="Add findings as accepted:")
+        queryset=Finding.objects.none(),
+        required=True,
+        label="Add findings as accepted:",
+        widget=TableCheckboxWidget(attrs={'size': 500})
```
Comment (Maffooch) [inline dojo/forms.py:783]:
```suggestion
        widget=TableCheckboxWidget(attrs={'size': 25})
```

File: dojo/widgets.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        self.page_number = kwargs.pop('page_number', 1)
+        super().__init__(*args, **kwargs)
+
+    def value_from_datadict(self, data, files, name):
+        selected_ids = data.getlist(name)
+        return [int(id) for id in selected_ids]
+
+    def render(self, name, value, attrs=None, renderer=None):
+        page_number = self.page_number
+        paginator = Paginator(self.findings, 500)  # 10 items per page
```
Comment (Maffooch) [inline dojo/widgets.py:22]:
```suggestion
        paginator = Paginator(self.findings, 25)  # 25 items per page
```

**Review by Maffooch — CHANGES_REQUESTED** [review]
If the screenshot is accurate in the description, the table would look a lot better if the label were above the tabel rather than to the left of it

File: dojo/widgets.py (hunk 3 above)
Comment (FelixHernandez) [reply dojo/widgets.py]: https://github.com/DefectDojo/django-DefectDojo/pull/9737/commits/1542e69fb6e34d4c6d67946e11bc358bac55e950

File: dojo/forms.py (hunk 2 above)
Comment (FelixHernandez) [reply dojo/forms.py]: https://github.com/DefectDojo/django-DefectDojo/pull/9737/commits/1542e69fb6e34d4c6d67946e11bc358bac55e950

File: dojo/engagement/views.py (hunk 1 above)
Comment (FelixHernandez) [reply dojo/engagement/views.py]: https://github.com/DefectDojo/django-DefectDojo/pull/9737/commits/1542e69fb6e34d4c6d67946e11bc358bac55e950

**Review by Maffooch — APPROVED** [review]

File: dojo/templates/dojo/view_risk_acceptance.html [hunk 4]
```diff
@@ -384,4 +385,37 @@ <h3>Notes</h3>
...
+                        render: function(data) {
+                            var date = new Date(data);
+                            var year = date.getFullYear();
+                            var month = (1 + date.getMonth()).toString().padStart(2, '0');
+                            var day = date.getDate().toString().padStart(2, '0');
+
+                            return year + '-' + month + '-' + day;
+                        }
+                    },
+                    { title: 'Active' }
```
Comment (cneill) [inline dojo/templates/dojo/view_risk_acceptance.html:415]: I'm not sure why this is happening, but the table headings don't move with the data columns when the window is resized, leading to confusing results.

Original:
Screenshot 2024-03-15 at 12 18 03

Resized:
Screenshot 2024-03-15 at 12 14 40

Additionally, these columns differ quite a bit from the standard Finding listing template:

Screenshot 2024-03-15 at 12 14 53

At a minimum I would:
* Swap Name and EPSS
* Use the same format for dates as used in the normal Finding list template

I'm also not sure ID needs to be present as a column here.

**Review by cneill — CHANGES_REQUESTED** [review]

File: dojo/templates/dojo/view_risk_acceptance.html (hunk 4 above)
Comment (FelixHernandez) [reply dojo/templates/dojo/view_risk_acceptance.html]: https://github.com/DefectDojo/django-DefectDojo/pull/9737/commits/562d1eccc08b885cb480e43b39e308a40aea90d4

**Review by cneill — APPROVED** [review]

**Review by grendel513 — APPROVED** [review]

**Pull request merged**